examples/preprocessing/plot_smoothing_noise_quantile_transform.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+========================================================
+Effect of smoothing noise when using QuantileTransformer
+========================================================
+
+This example shows the effect of applying a small noise before
+computing the quantiles in the QuantileTransformer. The parameter
+``smoothing_noise`` can be used if a constant feature value is
+predominant in the dataset. It will lead to a difficult interpretation
+when the quantiles computed are manually checked.
```

Comment:
I think this should be rephrased:

The parameter ``smoothing_noise`` can be used if some specific feature values
are repeated exactly many times to the point of being predominant in the dataset.

Without smoothing noise, the QuantileTransformer will map those values to some
arbitrary value: the highest quantile value for all the inputs with the same value.
While this is usually not an issue when QuantileTransformer is used as a
preprocessing transformer for a subsequent subsequent supervised estimator, it can
lead to surprising results when manually inspecting the transformed values (e.g. for
visualization or reporting).

The goal of the smoothing noise is to make it possible to map those repeated values
to some middle quantile value to make interpretation more intuitive as demonstrated
in the following.